File: petpals-pals/src/main/java/com/petpals/pals/domain/pals/model/PalMeasurement.java
package com.petpals.pals.domain.pals.model;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;

public record PalMeasurement(
        @DecimalMin(value = "0.0", inclusive = false) Double weight,
        @Min(value = 0) Double height
) {
}

File: petpals-pals/src/main/kotlin/com/petpals/domain/model/PalMeasurement.kt
package com.petpals.domain.model

import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.Min

data class PalMeasurement(
    val weight: @DecimalMin(value = "0.0", inclusive = false) Double,
    val height: @Min(value = 0) Double
)
